### PR TITLE
SelectPanel2: Add default Header and Footer, Replace SelectPanel.Heading

### PR DESCRIPTION
**Pull request by siddharthkp**
Related context:
- https://github.com/github/primer/pull/2490
- https://github.com/github/primer/issues/2396

&nbsp;

#### Part 1/2: Add default Header and Footer when not provided

This minimal example would render the fallback header and footer:

```jsx
<SelectPanel title="Select labels" onSubmit={onSubmit}>
  <SelectPanel.Button>Assign label</SelectPanel.Button>
  
  <SelectPanel.ActionList>
    {labels.map(label => (
      <ActionList.Item
        onSelect={() => onSelect(label.id)}
        selected={selectedLabelIds.includes(label.id)}
      >
        {label.name}
      </ActionList.Item>
    ))}
  </SelectPanel.ActionList>
</SelectPanel>
```

Minimal SelectPanel still renders a Header and Footer

&nbsp;

#### Part 2/2: Heading element is mandatory and not customisable

Related context:
- https://github.com/github/primer/issues/2578 (It's not, it should always render h1)

```jsx
// before:
<SelectPanel>
  <SelectPanel.Header>
    <SelectPanel.Heading>Select labels</SelectPanel.Heading>
  </SelectPanel.Header>
   ...
</SelectPanel>


// after:
<SelectPanel title="Select labels">
   ...
</SelectPanel>
```

### CODE REVIEW

**Comment by broccolinisoup** [comment]
@siddharthkp let me know when it is ready for review 🙌🏻

**Comment by siddharthkp** [comment]
@broccolinisoup Ready for you!

I have a follow up PR as well 🤭: https://github.com/primer/react/pull/3763

File: src/drafts/SelectPanel2/SelectPanel.stories.tsx [hunk 1]
```diff
@@ -428,29 +421,32 @@ export const TODO1Uncontrolled = () => {
...
-  return (
-    <>
-      <h1>TODO: Without any customisation</h1>
-      <p>Address after TODO: Uncontrolled</p>
-    </>
-  )
-}
-
-export const TODO4WithFilterButtons = () => {
+export const FWithFilterButtons = () => {
```
Comment (broccolinisoup) [inline src/drafts/SelectPanel2/SelectPanel.stories.tsx:424]: Your story names remind me of a PR that I opened a while ago and abandoned 😬 https://github.com/primer/react/pull/3588 I'll look into it 👀

**Review by broccolinisoup — APPROVED** [review]
🚀

**Comment by broccolinisoup** [comment]
>Canary Release is failing, but that is not related to the changes :(

Yeah super weird that happens occasionally and no idea why :( If you re-ran the job, it should go away hopefully 🤞🏻

File: src/drafts/SelectPanel2/SelectPanel.stories.tsx (hunk 1 above)
Comment (siddharthkp) [reply src/drafts/SelectPanel2/SelectPanel.stories.tsx:424]: 🤭 thank you! it's such a silly hack

**Pull request merged**